src/v2/coins/xlm.js

```diff
@@ -463,14 +475,22 @@ class Xlm extends BaseCoin {
       const minimumReserve = yield this.getMinimumReserve();
       const baseTxFee = yield this.getBaseTransactionFee();
       const recoveryAmount = walletBalance - minimumReserve - baseTxFee;
-
-      const txBuilder = new stellar.TransactionBuilder(account)
-      .addOperation(stellar.Operation.payment({
-        destination: params.recoveryDestination,
-        asset: stellar.Asset.native(),
-        amount: (recoveryAmount / this.getBaseFactor()).toString()
-      }))
-      .build();
+      const formattedRecoveryAmount = (recoveryAmount / this.getBaseFactor()).toString();
+
+      let txBuilder = new stellar.TransactionBuilder(account);
+
+      if (unfundedDestination) { // In this case, we need to create the account
+        txBuilder = txBuilder.addOperation(stellar.Operation.createAccount({
```

Comment:
You could bring the txBuilder.addOperation outside the if statement, since this is executed on both sides of the branch.